tests/test_git_colocated.rs

```diff
@@ -235,3 +235,104 @@ fn test_git_colocated_conflicting_git_refs() {
 fn get_log_output(test_env: &TestEnvironment, workspace_root: &Path) -> String {
     test_env.jj_cmd_success(workspace_root, &["log", "-T", "commit_id \" \" branches"])
 }
+
+#[test]
+fn test_git_colocated_unreachable_commits() {
+    let test_env = TestEnvironment::default();
+    let workspace_root = test_env.env_root().join("repo");
+    let git_repo = git2::Repository::init(&workspace_root).unwrap();
+
+    // Create an initial commit in Git
+    std::fs::write(workspace_root.join("file"), "contents").unwrap();
+    git_repo
+        .index()
+        .unwrap()
+        .add_path(Path::new("file"))
+        .unwrap();
+    let tree1_oid = git_repo.index().unwrap().write_tree().unwrap();
+    let tree1 = git_repo.find_tree(tree1_oid).unwrap();
+    let signature = git2::Signature::new(
+        "Someone",
+        "someone@example.com",
+        &git2::Time::new(1234567890, 60),
+    )
+    .unwrap();
+    let oid1 = git_repo
+        .commit(
+            Some("refs/heads/master"),
+            &signature,
+            &signature,
+            "initial",
+            &tree1,
+            &[],
+        )
+        .unwrap();
+    insta::assert_snapshot!(
+        git_repo.head().unwrap().peel_to_commit().unwrap().id().to_string(),
+        @"e61b6729ff4292870702f2f72b2a60165679ef37"
+    );
+
+    // Add a second commit in Git
+    std::fs::write(workspace_root.join("file"), "other contents").unwrap();
+    git_repo
+        .index()
+        .unwrap()
+        .add_path(Path::new("file"))
+        .unwrap();
+    let tree2_oid = git_repo.index().unwrap().write_tree().unwrap();
+    let tree2 = git_repo.find_tree(tree2_oid).unwrap();
+    let signature = git2::Signature::new(
+        "Someone",
+        "someone@example.com",
+        &git2::Time::new(1234567890, 62),
+    )
+    .unwrap();
+    let oid2 = git_repo
+        .commit(
+            Some("refs/heads/master"),
```

Comment:
I think you pass `None` here so the branch doesn't get updated. That should make the reset below unnecessary.

Reply:
Indeed. That's what I did first but I tried to be as close as possible to the reproducer mentionned in the issue. Done.